### DISCUSSION TITLE
RM68120 support

### DISCUSSION THREAD

**Discussion by Fischkopppp** [Bodmer/TFT_eSPI#1704] [New processor/display support or capability request]
Hi Bodmer, 
do you think you could add support for the RM68120 display driver (8bit parallel / SPI)? If I would send you this display:
https://de.aliexpress.com/item/32878013749.html
With best regards!

**Comment by Bodmer** [comment]
To do this I need the initialisation code sequence. Do you have a copy and have you you tested it to show it works?

**Comment by Fischkopppp** [comment]
I do not have this display. I thought about buying one. I asked the seller for the datasheet and for the initialisation code sequence. I will write again when I have the datasheet.

**Comment by Bodmer** [comment]
I found some init code here that might work.
https://github.com/espressif/esp-dev-kits/blob/master/esp32-s2-hmi-devkit-1/components/screen/controller_driver/rm68120/rm68120.c

An adaption of the example code there would also be a good starting point to test the display interface.

**Comment by Fischkopppp** [comment]
I was wondering what display Espressif is using on the HMI Dev kit. Now I know ;-)
If I buy myself a display, should I send one to you, too? Shipping will take a month.
You can send me a pm if you want: bikedynamos@googlemail.com

**Comment by Bodmer** [comment]
The HMI devkit code seems to support quite a few displays:
https://github.com/espressif/esp-dev-kits/tree/master/esp32-s2-hmi-devkit-1/components/screen/controller_driver

I am concerned that the RM68120 seems limited in vendors and it is an old chip design. So supplies may dry up. If you want touch then go for a capacitive one as the resistance screens scratch easily and are typically noisy and inaccurate. Capacitance ones often support multi-touch and swipe too.

**Comment by Bodmer** [comment]
I have added RM68120 support to v2.4.43.

It is competely untested but might work as the changes are not that significant, however I may have missed something...

Thanks, but no need to send me a display, I have one on order with the capacitance touch screen!  The aim is to replace my resistance based touch macro screen (based on [this code](https://github.com/DustinWatts/Pico-Matrix-Touch-Keyboard)) with a more pleasant to use multi-touch capacitance screen.

**Comment by Bodmer** [comment]
My RM68120  display arrived from China today. It will be a couple of weeks before I can try it with the library updates. I did not expect it to arrive quite so fast! Looks well made. I think the interface option links will need to be changed for 8 bit parallel operation.

**Reply by uldara1** [reply]
HI @Bodmer i have RM68120 display, but my display connected 16bit parallel with ESP32 S3,
can it use with your library ?
this is product that i use https://nl.aliexpress.com/item/1005003814428825.html?gatewayAdapt=glo2nld
github: https://github.com/W00ng/ESP32-S3-HMI-DevKit
[SCH-esp..80.pdf](https://github.com/lovyan03/LovyanGFX/files/8843326/SCH-esp.80.pdf)

**Comment by Bodmer** [comment]
The RM68120 support included in the library does not work. The only available datasheet has lots and lots of omissions, misleading information and errors!

I have managed to reverse engineer the interface from Espressif code and some guessing (!) and I now have an RP2040 Pico driving the display very nicely in 8 bit parallel mode. This code is not ready for release yet.

**Reply by david-res** [reply]
@Bodmer anything you can share on the RM68120 init code that is working?
Im writing a driver for a Teensy and had it working for a bit but now it seems dead.

**Reply by Bodmer** [reply]
The init sequence is here:
https://github.com/Bodmer/TFT_eSPI/blob/master/TFT_Drivers/RM68120_Defines.h
